crypto/merkle-tree/src/result.rs

```diff
@@ -10,3 +10,18 @@ pub enum Error {
     NotEnoughHashes,
     RootHashMismatch,
 }
+
+// Allows error bubbling up to string with '?' operator
```

Comment:
Is this standard? It seems wrong to convert errors into strings.

Reply:
Yes, this is an anti-pattern which we should not do. Instead we should implement the `Error` and `Display` traits.

https://doc.rust-lang.org/rust-by-example/error/multiple_error_types/define_error_type.html

https://doc.rust-lang.org/std/error/trait.Error.html